app/models/oai/figgy/oai_wrapper.rb

```diff
@@ -32,8 +36,22 @@ def sets
 
     private
 
+      def resource
+        __getobj__
+      end
+
       def decorated_resource
         __getobj__.decorate
       end
+
+      def mime_types
+        file_sets = decorated_resource.file_sets
```

Comment:
What if the resource doesn't respond to `file_sets`? They might all, I'm unsure.

Reply:
Seems like a perfectly reasonable case to hedge against.

Reply:
Added a test for this.